### PR TITLE
State routes

### PR DESCRIPTION
**Pull request by iygeal**
Check the state_views implementations

### CODE REVIEW

File: api/v1/views/states.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        if 'name' not in json_data:
+            abort(400, description="Missing name")
+
+        state = State(**json_data)
+        state.save()
+        return make_response(jsonify(state.to_dict()), 201)
+
+    except Exception:
+        abort(500, description="Internal server error")
```
Comment (loayalsaid1) [inline api/v1/views/states.py]: I think it was mentioned that you never raise this error explicitly..

This is only by the server under the hood

File: api/v1/views/states.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    """Method to create a State object"""
+    try:
+        json_data = request.get_json()
+        if not json_data:
+            abort(400, description="Not a JSON")
+
+        if 'name' not in json_data:
+            abort(400, description="Missing name")
+
+        state = State(**json_data)
```
Comment (loayalsaid1) [inline api/v1/views/states.py:49]: One important note is..

### What if the user adds an id, created_at, or updated_at property?

**Comment by loayalsaid1** [comment]
**Nice work indeed**

I think we use the leverage the <methods> parameter in the method... 

**meaning**... making as much functions as the routes we have..
and within each route.. we handle the different options or methods for it..

As the methods parameter is a list and not a string var or something..
it's made that we put all the options available for the route we are giving.


Nice Hard work.

File: api/v1/views/states.py (hunk 1 above)
Comment (iygeal) [reply api/v1/views/states.py]: I have commented out the try...except block

**Pull request closed without merging**